service/resource/legacyv2/keys.go

```diff
@@ -126,3 +126,7 @@ func networkNTPBlock(spec v1alpha1.FlannelConfigSpec) string {
 func networkTapName(spec v1alpha1.FlannelConfigSpec) string {
 	return "tap-" + clusterID(spec)
 }
+
+func serviceAccountNameForDeletion(spec v1alpha1.FlannelConfigSpec) string {
```

Comment:
Why is this function needed if it just delegates to `clusterID`? Could you just call `clusterID` instead of `serviceAccountNameForDeletion`?

Reply:
yes but I call it from different places (framework and create resource), so IMO If later we decide the name can be changed is easier. WDYT?

Reply:
IMO it's more clear to understand in place what name for deletion you are using (cluster ID) than having a wrapper around that function, i don't really see any value from `serviceAccountNameForDeletion`. Why do you think it would be easier to use the function if we decide later to change the name?

Reply:
Well imagine we leave ClusterId and in the future, we add another service account in the namespace for deletion. Then the hypothetical developer says ok let's rename the service account to be more specific than the clusterID, so he will change to something like ClusterID + 'some string', but he can make a mistake if he does not change in both places. In case the value comes from a function, you can change it, only changing the value in the function. It is a pattern I used in OOP but I can leave it as ClusterID if you think it has not sense

Reply:
IMO it makes more sense to not wrap clusterID if we don't add any value, I think now it makes things harder to understand. When the time comes to extend this we can think about how to make it properly, early optimizations usually don't help :)

Reply:
ok done :)